x-pack/plugin/esql/build.gradle

```diff
@@ -50,9 +50,9 @@ dependencies {
 }
 
 tasks.named("compileJava").configure {
-  options.compilerArgs.addAll(["-s", "src/main/generated"])
+  options.compilerArgs.addAll(["-s", "$projectDir/src/main/generated"])
   // IntelliJ sticks generated files here and we can't stop it....
-  exclude { it.file.toString().contains("src/main/generated-src/generated") }
+  exclude { it.file.toString().contains("$projectDir/src/main/generated-src/generated") }
```

Comment:
Yeah, the above change was never intended to change the _behavior_ only to avoid passing an absolute path into the compile task which reduces cache reuse. Since it seems the working directory of `javac` isn't predictable we'll need to provide another solution here, or as @nik9000 mentions, stop committing this stuff to source control. Although in either case we'll want to ensure we generate the source into a predictable location and not some folder under a compile worker working directory.

cc @breskeby